realm/realm-library/src/androidTest/java/io/realm/RealmRecyclerAdapterTest.java

```diff
@@ -0,0 +1,296 @@
+package io.realm;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.annotation.UiThreadTest;
+import android.support.test.rule.UiThreadTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.v7.widget.RecyclerView;
+import android.widget.FrameLayout;
+
+import io.realm.entities.AllTypes;
+import io.realm.entities.RealmRecyclerAdapter;
+import io.realm.entities.UnsupportedCollection;
+import io.realm.rule.TestRealmConfigurationFactory;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.anyInt;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(AndroidJUnit4.class)
+public class RealmRecyclerAdapterTest {
+
+    @Rule
+    public final UiThreadTestRule uiThreadTestRule = new UiThreadTestRule();
+    @Rule
+    public final TestRealmConfigurationFactory configFactory = new TestRealmConfigurationFactory();
+    private Context context;
+
+    private final static String FIELD_STRING = "columnString";
+    private final static int TEST_DATA_SIZE = 47;
+
+    private boolean automaticUpdate = true;
+    private Realm testRealm;
+
+    @Before
+    public void setUp() throws Exception {
+        context = InstrumentationRegistry.getInstrumentation().getContext();
+        RealmConfiguration realmConfig = configFactory.createConfiguration();
+        testRealm = Realm.getInstance(realmConfig);
+
+        populateTestRealm(testRealm, TEST_DATA_SIZE);
+    }
+
+    private void populateTestRealm(Realm testRealm, int objects) {
+        testRealm.beginTransaction();
+        for (int i = 0; i < objects; i++) {
+            AllTypes allTypes = testRealm.createObject(AllTypes.class);
+            allTypes.setColumnString("test data " + i);
+        }
+        testRealm.commitTransaction();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        testRealm.close();
+    }
+
+    @Test
+    public void constructor_testRecyclerAdapterParameterExceptions() {
+        RealmResults<AllTypes> resultList = testRealm.where(AllTypes.class).findAll();
+        try {
+            new RealmRecyclerAdapter(null, resultList, automaticUpdate);
+            fail("Should throw exception if context is null");
+        } catch (IllegalArgumentException ignore) {
+        }
+    }
+
+    @Test
+    @UiThreadTest
+    public void clear_testRemoveAllDataFromAdapter() {
```

Comment:
I would just call this method `clear` since this tests the default expected behavior